src/zjs_util.c

```diff
@@ -175,7 +175,7 @@ bool zjs_obj_get_double(jerry_value_t obj, const char *name, double *num)
     jerry_value_t value = zjs_get_property(obj, name);
     bool rval = false;
 
-    if (!jerry_value_has_error_flag(value)) {
+    if (!jerry_value_has_error_flag(value) && jerry_value_is_number(value)) {
```

Comment:
This does seem like a good idea, since usually people probably don't know for sure they will have a numeric property. But it changes the contract... so remove the part in the "requires:" line above that says "value name should exist as number".

The idea of "requires" is that you specify what is valid usage of the function and therefore don't have to check for invalid conditions within the function. Same change both places below.